crypto/s2n_certificate.h

```diff
@@ -67,6 +67,7 @@ int s2n_cert_chain_and_key_load_sans(struct s2n_cert_chain_and_key *chain_and_ke
 int s2n_cert_chain_and_key_matches_dns_name(const struct s2n_cert_chain_and_key *chain_and_key, const struct s2n_blob *dns_name);
 
 int s2n_cert_set_cert_type(struct s2n_cert *cert, s2n_pkey_type pkey_type);
+int s2n_cert_set_public_key(struct s2n_cert *cert, s2n_cert_public_key public_key);
```

Comment:
Where is this new method used?

Reply:
In s2n_certificate's `s2n_cert_chain_and_key_load_pem` api 

Usage : Store the parsed public key to s2n_cert

Reason : We want to cut down on parsing public_key every time during signature validation for the same cert. (i.e. executing expensive api -> `s2n_asn1der_to_public_key_and_type`)

Reply:
It's a one line method though. If it's not used outside of s2n_cert_chain_and_key_load_pem, why make it a separate method?

Reply:
Was trying to make it explicit and cleaner to read. 
But you are right, we are not using this anywhere else. Let me remove extra method.

Reply:
Done